src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,33 @@
+public class ArrayValueCalculator {
+
+    public static String doCalc(String stringArray[][]) throws ArraySizeException, ArrayDataException {
```

Comment:
`throws ArraySizeException, ArrayDataException` should be used if you wanna delegate error-handling to the end user, but it seems that you're validating the input on your own, so throws is redundant here, remove it.